### PR TITLE
feat: add auro-checkbox-group and auro-checkbox components

### PR DESCRIPTION
**Pull request by tsalright**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Issue:** https://github.com/AlaskaAirlines/ods-inputoptions/issues/26

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

This pull request is the initial build of the component with tests and documentation built into it.

## Type of change:

Please delete options that are not relevant.

- [x] New capability

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [ ] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

File: demo/index.html [hunk 1]
```diff
@@ -60,9 +71,184 @@ <h1>auro-checkbox</h1>
...
+              value="machines"
+            >machines</auro-checkbox>
+          </auro-checkbox-group>
+        </template>
+      </demo-snippet>
+
+      <p>Checkbox group with option set to <code>checked</code></p>
+      <demo-snippet>
+        <template>
+          <auro-checkbox-group label="Form label goes here">
```
Comment (tsalright) [inline demo/index.html:175]: This still uses the old label style. Should this be changed to the slot?

File: demo/index.html [hunk 2]
```diff
@@ -60,9 +71,184 @@ <h1>auro-checkbox</h1>
...
+              value="arts"
+            ></auro-checkbox>
+          </auro-checkbox-group>
+        </template>
+      </demo-snippet>
+
+      <p>Checkbox group set <code>disabled</code></p>
+      <demo-snippet>
+        <template>
+          <auro-checkbox-group label="Form label goes here" disabled>
```
Comment (tsalright) [inline demo/index.html:202]: Another possible slot update miss.

File: demo/index.html [hunk 3]
```diff
@@ -60,9 +71,184 @@ <h1>auro-checkbox</h1>
...
+          </auro-checkbox-group>
+        </template>
+      </demo-snippet>
+
+      <p>Checkbox group with error</p>
       <demo-snippet>
         <template>
-          <auro-checkbox cssClass="testClass">Hello World!</auro-checkbox>
+          <auro-checkbox-group
+            label="Form label goes here"
```
Comment (tsalright) [inline demo/index.html:230]: Another possible slot update miss

File: test/auro-checkbox.test.js [hunk 4]
```diff
@@ -1,28 +1,149 @@
...
+
+describe('auro-checkbox-group', () => {
+  it('has the expected properties', async () => {
+    const expectedError = "Expected error message";
 
-describe('auro-checkbox', () => {
-  it('sets the CSS class on auro-checkbox > div element', async () => {
     const el = await fixture(html`
-      <auro-checkbox cssclass="testClass"></auro-checkbox>
+      <auro-checkbox-group
```
Comment (tsalright) [inline test/auro-checkbox.test.js:16]: Should the label change be tested in here as well?

Comment (blackfalcon) [reply test/auro-checkbox.test.js:16]: The purpose of a test is to evaluate the code we write. We are not responsible for the functionality of the `<slot>` element, so it's my opinion that we don't need to test that feature.

**Pull request merged**